Tests/BootCamp.Chapter.Tests/GameComponents/DeckTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using BootCamp.Chapter.Gambling;
+using Xunit;
+
+namespace BootCamp.Chapter.Tests.GameComponents
+{
+    public class DeckTests
+    {
+        private List<Card> _someCards = new List<Card>
+        {
+            new Card(Card.Suites.Clubs, Card.Ranks.Ace),
+            new Card(Card.Suites.Clubs, Card.Ranks.Two),
+            new Card(Card.Suites.Clubs, Card.Ranks.Three)
+        };
+
+        [Fact]
+        public void Deck_Given_Null_Cards_Throws_ArgumentNullException()
+        {
+            void NullDeck() => CreateDeck(null);
+
+            Assert.Throws<ArgumentNullException>(NullDeck);
+        }
+
+        [Fact]
+        public void Deck_DrawAt_Returns_Expected_Card()
```

Comment:
Doesn't explain well enough what is tested. `Deck` is not needed, it's DeckTests after all.

Reply:
DrawAt_Given_2Cards_When_Index1_Returns_2ndCard.
However, you should test edge cases: 1st, middle, last card:
```
[Theory]
[...]
DrawAt_Given_3Cards_Returns_Expected()
```